README.md

```diff
@@ -1,28 +1,121 @@
+
+```
+git clone https://github.com/capsule-corp-ternoa/ternoa-subql-dictionary
+```
+
+#### 2. Change Directory
+
+Change the Directory for the desired results : 
+```
+cd ternoa-subql-dictionary
+```
+
+#### 3. Select Testnet version
+
+Select the appropriate testnet version which corresponds to the Indexer version you are using to ensure compatibility :
+
+```
+git checkout v40/testnet
+# The indexer and dictionary should be on same version.
 ```
 
-### 2. Generate types
-```shell
+#### 4. Install Dependencies
+
+Install the required dependencies for the project using :
+
+```
+yarn install
+```
+#### 5. Generate Code from GraphQL
+
+Generate code from your GraphQL schema and Operations :
+
+```
 yarn codegen
 ```
 
-### 3. Build
-```shell
+#### 6. Build your Implementation
+
+Create an Executable version of your project :
+    
+```
 yarn build
 ```
 
-### 4. Run locally
-```shell
-yarn start:docker
+#### 7. Docker pull
+
+Pull the latest versions of the Docker image using :
+
+```
+docker-compose pull
+```
+
+#### 8. Run 
+
+Run your compiled app with Docker using :
+
+```
+docker-compose up
+```
+
+**Wait a couple of seconds for the indexing to start, after that you can access the Blockchain data in your [local](http://localhost:3000/) GraphQL playground**
+
+
+## Using the Dictionary
+
+You can significantly improve the indexation performance by using a dictionary endpoint instead of targeting the blockchain directly to gather data.
+
+Just make sure to select the correct endpoints for the desired networks :
+
+As you can see that the endpoints for Networks are structured as `Wss://xxxxxxxx.ternoa.com` and `https://dictionary-xxxxxxxx.ternoa.com` which is replaced by the `Network designation`.
+
+The most famous network designations being `Mainnet`, `Alphanet`, `Chaosnet`, `Testnet` and various sub versions of the existing networks.
+
+#### Examples -
+
+For use on the `Testnet` :
+```
+  genesisHash: '0xd9adfc7ea82be63ba28088d62b96e9270ad2af25c962afc393361909670835b2' 
+  endpoint: 'wss://testnet.ternoa.com'
+  dictionary: 'https://dictionary-testnet.ternoa.dev/'
+  chaintypes:
+    file: ./types.json
+```
+
+For use on the `Mainnet` :
 ```
+  genesisHash: '0xd44bcfb0e98da45ace37e4df8469e5dbba8c4fc5449acda24c50cea6f5f2ca99'
+  endpoint: 'wss://mainnet.ternoa.com'
+  dictionary: 'https://dictionary-mainnet.ternoa.dev/'
+  chaintypes:
+    file: ./types.json
+```
+
+The same goes for the `Chaos` network or its feature staging area `Staging.Chaos` 
+
+The network endpoints for the sub nets stay the same whereas the dictionary endpoint varies slightly like `Staging.Chaos` or `Dev.Chaos`, only requiring the specific versions of a network. 
+
+For example, the Dictionary endpoint for `Staging.Chaos` being `https://dictionary-staging.ternoa.dev/` and `https://dictionary-dev.ternoa.dev/` for `Dev.Chaos`.
+    
+**Learn more about the different network versions over at `Ternoa-Fundamentals/Networks`**
+
+## Using Dictionary as an Explorer
```

Comment:
In this section you can remove the 2 paragraphs and just go with:

[Ternoa Explorer](https://explorer.ternoa.com/) is an example of project relaying on Ternoa Dictionnary. It enables you to visualize the on-chain activity, explore public addresses and their transaction histories, gather information about individual blocks and much more.

Reply:
Done